odoo/fields.py

```diff
@@ -2235,6 +2235,9 @@ class One2many(_RelationalMulti):
 
         The attributes ``comodel_name`` and ``inverse_name`` are mandatory except in
         the case of related fields or field extensions.
+
+        Using ``related`` to refer to another One2many or Many2many field is not
+        implemented and will result in undefined behavior.
```

Comment:
Maybe we should trigger a warning if that isn't already the case? /cc @rco-odoo

Reply:
I didn't see one when I encountered the issue in master recently, and I agree a warning (or even an error) would be a good thing.

Reply:
I don't know any related `o2m `in the code currently (and indeed, it isn't supported), but I know at least one related `m2m `in the code ...  This field isn't stored, but is still basically supported.

If we want a warning/error, it should be more targeted than simply targeting all related `x2m` ;).

Reply:
@Feyensv Could you paste a link to that `m2m` related? I had understood from mat that that wouldn't work in the ORM. It will appear to work if there's only 1 instance you're pointing to, but it falls apart if there's multiple (at least in the chained `o2m` cases I tested).

Reply:
@Feyensv I tested and reproduced with a `m2m` field using `related` to chain through another `m2m` field. 

Here's the models I used:

```python
from odoo import models, fields, api

class A(models.Model):
    _name = 'related_m2m.a'

    name_a = fields.Char()
    b_ids = fields.Many2many('related_m2m.b')
    c_ids = fields.Many2many(related='b_ids.c_ids')


class B(models.Model):
    _name = 'related_m2m.b'

    name_b = fields.Char()
    c_ids = fields.Many2many('related_m2m.c')


class C(models.Model):
    _name = 'related_m2m.c'

    name_c = fields.Char()
```

A has a `m2m` to B and B has a `m2m` to C. A defines an additional `related` `m2m` field to access linked C instances through B, but if you access it you will only get results for the first B instance:

```python
In [1]: a1 = self.env['related_m2m.a'].browse(1)

In [2]: a1.b_ids
Out[2]: related_m2m.b(2, 3)

In [3]: a1.b_ids[0]
Out[3]: related_m2m.b(2,)

In [4]: a1.b_ids[0].c_ids
Out[4]: related_m2m.c(4, 5)

In [5]: a1.b_ids[1].c_ids
Out[5]: related_m2m.c(6, 7)

In [6]: a1.c_ids
Out[6]: related_m2m.c(4, 5)
```

Underlying data: https://gist.github.com/raf-odoo/3d16924e858a2d0f73fc814319c68f94

Reply:
What I meant was sthg like `related_m2m = fields.Many2many(related="m2o_field.m2m_field")`, not `m2m_field.m2m_field` which are indeed not supported, but your comment doesn't leave this possibility open.

Didn't notice your question from yesterday: here is an example of my usecase ;) : https://github.com/odoo/odoo/blob/391c5292295507bd5451308ba227bd45b7d807be/addons/sale_product_configurator/models/sale_order_line.py#L11

Reply:
Aha, I see your point. Indeed, using `related` to refer to a `m2m` through a `m2o` will work and my current comment excludes that. I'll update it to be more precise.

Reply:
I updated it to be more precise, added a little example and put a warning box around it for emphasis.